src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -392,10 +501,81 @@ def _validate_vmss_create_load_balancer(namespace):
 
 def process_vmss_create_namespace(namespace):
     validate_location(namespace)
-    _validate_vm_create_storage_profile(namespace)
+    _validate_vm_create_storage_profile(namespace, for_scale_set=True)
     _validate_vmss_create_load_balancer(namespace)
     _validate_vm_create_vnet(namespace)
-    _validate_vm_create_public_ip(namespace)
+    _validate_vmss_create_subnet(namespace)
+    _validate_vmss_create_public_ip(namespace)
     _validate_vm_create_auth(namespace)
 
 # endregion
+
+# region disk, snapshot, image validators
+
+
+def validate_vm_disk(namespace):
+    namespace.disk = _get_ressource_id(namespace.disk, namespace.resource_group_name,
```

Comment:
no, it is to convert a name to resource id, so that the command can accept both format